tools/pbstate/src/pbstate.c

```diff
@@ -155,29 +157,30 @@ int pbstate_load(const char *p_device,
 
     fd = open(primary_device, O_RDONLY);
 
-    if (fd == -1) {
-        LOG("Error: Could not open '%s'\n", primary_device);
-        return -1;
-    }
-
-    if (read_config(fd, &config) != 0) {
-        LOG("Could not read primary config data\n");
+    if (fd != -1) {
+        if (read_config(fd, &config) != 0) {
+            LOG("Could not read primary config data\n");
+        }
+        close(fd);
+    } else {
+        err = -errno;
+        LOG("Error: Could not open '%s' (%i)\n", primary_device, err);
+        return err;
     }
 
-    close(fd);
     fd = open(backup_device, O_RDONLY);
 
-    if (fd == -1) {
-        LOG("Error: Could not open '%s'\n", primary_device);
-        return -1;
-    }
-
-    if (read_config(fd, &config_backup) != 0) {
-        LOG("Could not read backup config data\n");
+    if (fd != -1) {
```

Comment:
As a general style I prefer (and most code seems to be) in a style where the success case is as little indented as possible, that is you try as much as possible to have
```
if (error) {
  .. return or break
}

<success case>
```
If nothing else just to keep consistency, consider changing.

Reply:
Ah, that is my misunderstanding of the logic in the method.